client/src/shared/components/workflow/WorkflowDialog.tsx

```diff
@@ -18,9 +18,17 @@ import {IntegrationWorkflowKeys} from '@/ee/shared/queries/embedded/integrationW
 import {Workflow} from '@/shared/middleware/platform/configuration';
 import {ProjectWorkflowKeys} from '@/shared/queries/automation/projectWorkflows.queries';
 import {UseMutationResult, UseQueryResult, useQueryClient} from '@tanstack/react-query';
-import {ReactNode, useEffect, useState} from 'react';
+import {KeyboardEvent, ReactNode, useEffect, useRef, useState} from 'react';
 import {useForm} from 'react-hook-form';
 
+const handleEnterKeyDown = (handler: () => void) => {
+    return (e: KeyboardEvent) => {
+        if (e.key === 'Enter' && !e.shiftKey) {
+            handler();
+        }
+    };
+};
+
```

Comment:
`e` -> `event`, ne volimo skraćenice.

`=>` je implicitni return, tako da su vitičaste zagrade i `return` višak.

Također, `handleEnterKeyDown` funkcija vraća anonymous funkciju (callback) `(e) => {...}` koja poziva argument koji je `handleEnterKeyDown` primila šta je jedna funkcija viška.

Uglavnom malo je prekompleksna implementacija s obzirom na to šta želimo da napravi.

Nama triba nešto drastično jednostavnije, doslovno je dovoljno stavit `onKeyDown={() => handleSubmit(saveWorkflow)}` umisto `onKeyDown={handleEnterKeyDown(submitWorkflowToSaving)}` i ne triba nam ništa drugo - tvoja implementacija dodaje 2 sloja abstrakcije koji nam ne trebaju.

Ref + focus logika može ostat, samo promijeni ime `labelRef` u `labelInputRef`

Reply:
Sorry, vidin da san ja napravija jednu grešku - sa mojim prijedlogom ne radi submitanje na enter. U sljedećem komentaru ću ti napisat ispravak svog rješenja